### PR TITLE
Auto multi validation

### PR DESCRIPTION
**Pull request by mariuszmalek**
I created this branch to have every validation in one place by class name.

Example use:

```
$array = EmailValidatorFactory::create("my@email.com");
```

### CODE REVIEW

**Comment by mariuszmalek** [comment]
Thanks to this, I can now easily handle all the information and save it in the database.

File: src/EmailValidatorFactory.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        RFCValidation::class,
+        NoRFCWarningsValidation::class,
+        DNSCheckValidation::class
+    ];
+    
+    /**
+     * @param string $emailAddress
+     * @return array
+     */
+    public static function create(string $emailAddress): array
```
Comment (egulias) [inline src/EmailValidatorFactory.php:25]: Which are the reasons behing making it static? It prevents being injected as dependency.

File: src/EmailValidatorFactory.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        $validator = new EmailValidator();
+        $result = [];
+
+        foreach (self::$defaultValidators as $key => $val) {
```
Comment (egulias) [inline src/EmailValidatorFactory.php:30]: You could just use ´MultipleValidationWithAnd´ at construction time. And remove line 15.

File: src/EmailValidatorFactory.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+use Egulias\EmailValidator\EmailValidator;
+use Egulias\EmailValidator\Validation\DNSCheckValidation;
+use Egulias\EmailValidator\Validation\NoRFCWarningsValidation;
+use Egulias\EmailValidator\Validation\RFCValidation;
+
+class EmailValidatorFactory
+{
+    /** @var [] */
+    protected static array $defaultValidators = [
```
Comment (egulias) [inline src/EmailValidatorFactory.php:15]: Are you expecting or encouraging extension? This prevents it. There is also no entry point to add additional/other validators than this.

You can do it via constructor arguments with sensible defaults (is like your idea here, but with simple extension point)

File: src/EmailValidatorFactory.php (hunk 1 above)
Comment (egulias) [reply src/EmailValidatorFactory.php:25]: Also, you are returning the lists of results, not a validator ready for use. I'd suggest to change the name to something like `validationsResults` or similar.

File: src/EmailValidatorFactory.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    public static function create(string $emailAddress): array
+    {
+        $validator = new EmailValidator();
+        $result = [];
+
+        foreach (self::$defaultValidators as $key => $val) {
+            $result[get_class(new $val)] = $validator->isValid($emailAddress, new $val);
+        }
+
+        return $result;
```
Comment (egulias) [inline src/EmailValidatorFactory.php:34]: Is the validator's `Result` type not useful? Maybe adding which validation is giving which result would be useful?

**Review by egulias — CHANGES_REQUESTED** [review]
Hey. Thanks for your patience. Leaving you here some comments.

**Comment by egulias** [comment]
Closing. Branch 3.x is no longer maintained.

**Pull request closed without merging**